src/Factory/EntryFactory.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Copyright 2020 Jesse Rushlow - Geeshoe Development
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace Geeshoe\Atom\Factory;
+
+use Geeshoe\Atom\Exception\FactoryException;
+use Geeshoe\Atom\Model\Entry;
+
+/**
+ * Class EntryFactory
+ *
+ * @package Geeshoe\Atom\Factory
+ * @author  Jesse Rushlow <jr@geeshoe.com>
+ */
+class EntryFactory
+{
+    /**
+     * @param string             $id
+     * @param string             $title
+     * @param \DateTimeInterface $updated
+     * @return Entry
+     */
+    public static function createEntry(string $id, string $title, \DateTimeInterface $updated): Entry
+    {
+        if (self::validateRequiredEntryElements($id, $title)) {
+            return new Entry($id, $title, $updated);
+        }
+
+        throw FactoryException::requiredException();
+    }
+
+    /**
+     * @param string $id
+     * @param string $title
+     * @return bool
+     */
+    protected static function validateRequiredEntryElements(string $id, string $title): bool
+    {
+        return !empty($id) && !empty($title);
+    }
```

Comment:
As atom:entry & atom:feed elements have the same required attributes, refactor to validate using a common trait or validator service.